core/src/Kokkos_LogicalSpaces.hpp

```diff
@@ -0,0 +1,433 @@
+    underlying_allocator.impl_deallocate(
+        arg_label, arg_alloc_ptr, arg_alloc_size, arg_logical_size, arg_handle);
+  }
+};
+}  // namespace Experimental
+}  // namespace Kokkos
+
+//----------------------------------------------------------------------------
+
+namespace Kokkos {
+
+namespace Impl {
+
+template <typename BaseSpace, typename DefaultExecutionSpace, class Namer,
+          typename OtherSpace>
+struct MemorySpaceAccess<Kokkos::Experimental::LogicalMemorySpace<
+                             BaseSpace, DefaultExecutionSpace, Namer, true>,
+                         OtherSpace> {
+  enum { assignable = MemorySpaceAccess<BaseSpace, OtherSpace>::assignable };
+  enum { accessible = MemorySpaceAccess<BaseSpace, OtherSpace>::accessible };
+  enum { deepcopy = MemorySpaceAccess<BaseSpace, OtherSpace>::deepcopy };
+};
+
+template <typename BaseSpace, typename DefaultExecutionSpace, class Namer,
+          typename OtherSpace>
+struct MemorySpaceAccess<OtherSpace,
+                         Kokkos::Experimental::LogicalMemorySpace<
+                             BaseSpace, DefaultExecutionSpace, Namer, true>> {
+  enum { assignable = MemorySpaceAccess<OtherSpace, BaseSpace>::assignable };
+  enum { accessible = MemorySpaceAccess<OtherSpace, BaseSpace>::accessible };
+  enum { deepcopy = MemorySpaceAccess<OtherSpace, BaseSpace>::deepcopy };
+};
+
+template <typename BaseSpace, typename DefaultExecutionSpace, class Namer>
+struct MemorySpaceAccess<Kokkos::Experimental::LogicalMemorySpace<
+                             BaseSpace, DefaultExecutionSpace, Namer, true>,
+                         Kokkos::Experimental::LogicalMemorySpace<
+                             BaseSpace, DefaultExecutionSpace, Namer, true>> {
+  enum { assignable = true };
+  enum { accessible = true };
+  enum { deepcopy = true };
+};
+
+template <typename BaseSpace, typename DefaultExecutionSpace, class Namer>
+struct MemorySpaceAccess<Kokkos::AnonymousSpace,
+                         Kokkos::Experimental::LogicalMemorySpace<
+                             BaseSpace, DefaultExecutionSpace, Namer, true>> {
+  enum { assignable = true };
+  enum { accessible = true };
+  enum { deepcopy = true };
+};
+
+}  // namespace Impl
+
+}  // namespace Kokkos
+
+//----------------------------------------------------------------------------
+
+namespace Kokkos {
+
+namespace Impl {
+template <class BaseSpace, class DefaultExecutionSpace, class Namer,
+          bool SharesAccessSemanticsWithBase>
+class SharedAllocationRecord<
+    Kokkos::Experimental::LogicalMemorySpace<
+        BaseSpace, DefaultExecutionSpace, Namer, SharesAccessSemanticsWithBase>,
+    void> : public SharedAllocationRecord<void, void> {
+ private:
+  using SpaceType = Kokkos::Experimental::LogicalMemorySpace<
+      BaseSpace, DefaultExecutionSpace, Namer, SharesAccessSemanticsWithBase>;
+  using RecordBase = SharedAllocationRecord<void, void>;
+
+  friend SpaceType;
```

Comment:
Why?  It does not seem like you are accessing anything private nor protected from `LogicalMemorySpace`.

Reply:
This mirrors all the other implementations, I think, but I'll double check

Reply:
Turns out this isn't necessary